ui/app/registration/controllers/patientCommonController.js

```diff
@@ -17,7 +17,186 @@ angular.module('bahmni.registration')
+                            identifier.generate();
+                            if (!identifierMatch) {
+                                extensionParam.addressMap !== null ? updatePatientAddress(patient.address[0], extensionParam.addressMap) : {};
+                                changePatientDetails(patient);
+                                identifierMatch = true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            function updatePatientAddress (address, addressMap) {
+                for (var key in addressMap) {
+                    if (key === "line") {
+                        $scope.patient.address[addressMap[key]] = address[key] !== null ? address[key].join(" ") : "";
+                    } else { $scope.patient.address[addressMap[key]] = address[key] !== null ? address[key] : ""; }
+                }
+            }
+
+            function updatePatientName (name) {
+                $scope.patient.givenName = name.givenName[0];
+                $scope.patient.middleName = name.givenName.length > 1 ? name.givenName[1] : "";
+                $scope.patient.familyName = name.familyName;
+            }
+
+            function changePatientDetails (changedDetails) {
+                for (var key in changedDetails) {
+                    switch (key) {
+                    case 'names':
+                        for (var i = 0; i < changedDetails.names.length; i++) {
+                            if (changedDetails.names[i].use === "preferred") {
+                                updatePatientName(changedDetails.names[i]);
+                                break;
+                            }
+                        }
+                        updatePatientName(changedDetails.names[0]);
+                        break;
+                    case 'gender':
+                        $scope.patient.gender = changedDetails.gender;
+                        break;
+                    case 'contactPoint':
+                        for (var i = 0; i < changedDetails.contactPoint.length; i++) {
+                            var contact = changedDetails.contactPoint[i];
+                            if (contact.system === "phone") { $scope.patient.primaryContact = contact.value; }
+                        }
+                        break;
+                    default:
+                        var age = calculateAge('01/01/' + changedDetails.birthDate);
+                        $scope.patient.age.years = age.years;
+                        $scope.patient.age.months = age.months;
+                        $scope.patient.age.days = age.days;
+                        $scope.patient.calculateBirthDate();
+                        break;
+                    }
+                }
+            }
+
+            $scope.closeIdentifierPopup = function () {
+                $scope.showIframe = false;
+            };
+
+            function calculateAge (birthDate) {
```

Comment:
there should be already a function that calculates patient age from a date. Check createPatientContrroller or editPatientController .. you can pull that out to a common utils

Reply:
reused the existing function